basic-search-java/src/test/java/com/mydomain/example/SystemTestExample.java

```diff
@@ -0,0 +1,151 @@
+package com.mydomain.example;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.nio.charset.StandardCharsets;
+import java.util.HashSet;
+import java.util.Iterator;
+
+import static java.net.URLEncoder.encode;
+import static java.net.http.HttpRequest.BodyPublishers.ofInputStream;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+
+/**
+ * System tests - demonstrates:
+ * <ol>
+ *     <li>How to set up the endpoint to the test instance</li>
+ *     <li>Feed to the test instance</li>
+ *     <li>Query the test instance</li>
+ *     <li>Visit the test instance</li>
+ * </ol>
+ */
+public class SystemTestExample {
+
+    private String getTestEndpoint() {
+        if (false) {   // something here when running in pipeline
+            return "";
+        }
+        return "http://endpoint:10000/";
+    }
+
+
+    //@BeforeEach
+    public void init() throws Exception {
+        removeAllDocuments("document/v1/music/music/docid/");
+    }
+
+
+    //@Test  // Another annotation here so it will be run in pipeline
+    public void simpleGETSearchTest() throws Exception {
+
+        assertEquals(0, getSearchResults(searchDocumentsGET(getTestEndpoint()
+                + "/search/?yql="
+                + encode("select * from sources * where sddocname contains \"music\";", StandardCharsets.UTF_8))),
+                "Expected 0 documents before feeding");
+
+        postDocument(getTestEndpoint() + "document/v1/music/music/docid/1", "/music-data-1.json");
```

Comment:
Also something I think we may end up providing in the library. It's fine that's user-implemented for now. 
We should showcase document updates as well.